src/luacov/reporter.lua

```diff
@@ -43,54 +59,50 @@ local function check_long_string(line, in_long_string, ls_equals, linecount)
-   { true, fixup "do" },       -- Single "do"
-   { true, fixup "if" },       -- Single "if"
-   { true, fixup "then" },     -- Single "then"
-   { true, fixup "while true do" }, -- "while true do" generates no code
-   { true, fixup "if true then" }, -- "if true then" generates no code
-   { true, fixup "local <IDS>" }, -- "local var1, ..., varN"
-   { true, fixup "local <IDS>=" }, -- "local var1, ..., varN ="
-   { true, fixup "local function(<ARGS>)" }, -- "local function(arg1, ..., argN)"
-   { true, fixup "local function <ID>(<ARGS>)" }, -- "local function f (arg1, ..., argN)"
+local any_hits_exclusions = {
+   "", -- Empty line
+   fixup "end[,;%s%)]*", -- Single "end"
+   "else", -- Single "else"
+   "repeat", -- Single "repeat"
+   "do", -- Single "do"
+   "if", -- Single "if"
+   "then", -- Single "then"
+   fixup "while true do", -- "while true do" generates no code
+   fixup "if true then", -- "if true then" generates no code
+   fixup "local <IDS>", -- "local var1, ..., varN"
+   fixup "local <IDS>=", -- "local var1, ..., varN ="
+   fixup "local function(<ARGS>)", -- "local function(arg1, ..., argN)"
+   fixup "local function <ID>(<ARGS>)", -- "local function f (arg1, ..., argN)"
 }
 
 --- Lines that are only excluded from accounting when they have 0 hits
-local hit0_exclusions = {
-   { true, "[%w_,='\" ]+," }, -- "var1 var2," multi columns table stuff
-   { true, fixup "<FIELDNAME>=.+[,;]" }, -- "[123] = 23," "['foo'] = "asd","
-   { true, fixup "<ARGS>*function(<ARGS>)" }, -- "1,2,function(...)"
-   { true, fixup "return <ARGS>*function(<ARGS>)" }, -- "return 1,2,function(...)"
-   { true, fixup "return function(<ARGS>)" }, -- "return function(arg1, ..., argN)"
-   { true, fixup "function(<ARGS>)" }, -- "function(arg1, ..., argN)"
-   { true, fixup "local <ID>=function(<ARGS>)" }, -- "local a = function(arg1, ..., argN)"
-   { true, fixup "<FULLID>=function(<ARGS>)" }, -- "a = function(arg1, ..., argN)"
-   { true, fixup "break" }, -- "break" generates no trace in Lua 5.2
-   { true, "{" }, -- "{" opening table
-   { true, "}" }, -- "{" closing table
-   { true, fixup "})" }, -- function closer
-   { true, fixup ")" }, -- function closer
+local zero_hits_exclusions = {
+   "[%w_,='\"%s]+,", -- "var1 var2," multi columns table stuff
+   fixup "<FIELDNAME>=.+[,;]", -- "[123] = 23," "['foo'] = "asd","
+   fixup "<ARGS>*function(<ARGS>)", -- "1,2,function(...)"
+   fixup "return <ARGS>*function(<ARGS>)", -- "return 1,2,function(...)"
+   fixup "return function(<ARGS>)", -- "return function(arg1, ..., argN)"
+   fixup "function(<ARGS>)", -- "function(arg1, ..., argN)"
+   fixup "local <ID>=function(<ARGS>)", -- "local a = function(arg1, ..., argN)"
+   fixup "<FULLID>=function(<ARGS>)", -- "a = function(arg1, ..., argN)"
+   "break", -- "break" generates no trace in Lua 5.2+
+   "{", -- "{" opening table
+   "}?[%s%)]*", -- optional "{" closing table, possibly with several closing parens
```

Comment:
this matches the empty string ( "}?" accepts "" and "[_anything_]*" accepts "" too)